### PR TITLE
Response

### PR DESCRIPTION
**Pull request by edesilets**
Used yesterdays session as a guide.

comfort 3

### CODE REVIEW

**Comment by GA-MEB** [comment]
Why are you trying to merge into 010/solution?

File: diagnostic.js [hunk 1]
```diff
@@ -29,21 +29,39 @@ mongoose.Promise = global.Promise;
 mongoose.connect('mongodb://localhost/mongoose-crud');
 
 /// ADD YOUR CODE BELOW
+const controller = require('lib/wiring/controller');
+const models = require('app/models');
```
Comment (MattB-GA-Boston) [inline diagnostic.js]: These two things are constructs from Antony's Express template - they are _not_ part of Mongoose and do not apply here.

File: diagnostic.js [hunk 2]
```diff
@@ -29,21 +29,39 @@ mongoose.Promise = global.Promise;
...
+const Art = models.art;
```
Comment (GA-MEB) [inline diagnostic.js:34]: These two things are constructs from Antony's Express template - they are _not_ part of Mongoose and do not apply here.

To require `Art`, you'd need to write

```
const Art = require('./models/art.js');
```

File: diagnostic.js [hunk 3]
```diff
@@ -29,21 +29,39 @@ mongoose.Promise = global.Promise;
...
 
-const create = (name, description, startYear, endYear) => {};
+const create = (name, description, startYear, endYear) => {
+  let art = Object.assign({ _owner: req.currentUser._id }, req.body.art);
```
Comment (GA-MEB) [inline diagnostic.js:37]: This isn't Express: `req` and `res` aren't defined here.
Additionally, there are no users here, so no owners.

File: models/art.js [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+// - An 'ancestor', another art movement that this movement descends from.
+
+const mongoose = require('mongoose');
+
+const artSchema = new mongoose.Schema({
+  name: {
+    type:String,
+    required:true
+  },
+  description: String,
```
Comment (GA-MEB) [inline models/art.js:16]: Missing a `maxlength` constraint:

> "A description (no longer than 200 characters)"

File: models/art.js [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+  startingYear: String,
+  endingYear: String,
+  _owner: {
+    type: mongoose.Schema.Types.ObjectId,
+    ref: 'User',
+  },
```
Comment (GA-MEB) [inline models/art.js:22]: Should be `_ancestor` and refer to another Movement (or, as you've named it Art).

File: models/art.js [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+}, {
+  timestamps: true,
+  toJSON: { virtuals: true },
+});
+
+exampleSchema.virtual('ancestor').get(function length() {
+  return this.text.length;
+});
```
Comment (GA-MEB) [inline models/art.js:30]: See note above.

**Comment by GA-MEB** [comment]
Unfortunately, although your model stuff was mostly right, all of your CRUD actions are wrong because you wrote them as if they were part of an Express app.

I'd strongly recommend that you resubmit and give this one another shot.

**Pull request closed without merging**